workspaces/adventure-pack/goodies/typescript/Iterator.from/index.ts

```diff
@@ -0,0 +1,44 @@
+import { iteratorPrototype } from "../Iterator.prototype";
+import "../Iterator.prototype.toIterable";
+
+declare global {
+  interface IteratorConstructor {
+    from<T>(
+      object: Iterator<T> | Iterable<T> | { next(): IteratorResult<T> },
+    ): IterableIterator<T>;
+  }
+
+  const Iterator: IteratorConstructor;
+}
+
+(globalThis as Record<string, unknown>).Iterator ??= {};
+
+(globalThis as Record<string, unknown>).Iterator = {
+  from<T>(
+    object: Iterator<T> | Iterable<T> | { next(): IteratorResult<T> },
+  ): IterableIterator<T> {
+    if (typeof (object as Iterable<T>)[Symbol.iterator] === "function") {
+      const iterable = object as Iterable<T>;
+      const iterator = iterable[Symbol.iterator]();
+      return iteratorPrototype.toIterable.call(iterator) as IterableIterator<T>;
+    }
```

Comment:
Fun question for you: why do we have to do `iteratorFactory.call(object)` and can't just do `iteratorFactory()`?

```suggestion
    const iteratorFactory = (object as Iterable<T>)[Symbol.iterator];
    if (typeof iteratorFactory === "function") {
      return toIterable.call(iteratorFactory.call(object));
    }
```

Reply:
Yeah this is because of the binding of `this` right, and we need to bind it to the `object`?